src/inference/src/dev/icompiled_model.cpp

```diff
@@ -48,34 +48,47 @@ ov::ICompiledModel::ICompiledModel(const std::shared_ptr<const ov::Model>& model
             }
         }
 
-        if (add_operation_names) {
-            for (const auto& param : model->get_parameters()) {
-                const auto& param_name = param->get_friendly_name();
+        for (const auto& param : model->get_parameters()) {
+            const auto& param_name = param->get_friendly_name();
+            auto new_param = ov::as_type_ptr<ov::op::v0::Parameter>(param->copy_with_new_inputs({}));
+            new_param->set_friendly_name(param_name);
+            if (add_operation_names) {
                 OPENVINO_ASSERT(!m_plugin->is_new_api() || leaf_names.find(param_name) == leaf_names.end() ||
                                     param->output(0).get_names().find(param_name) != param->output(0).get_names().end(),
                                 "Model operation names have collisions with tensor names.",
                                 " Please use MO to generate new IR version, it should allow to avoid the issue");
                 leaf_names.insert(param_name);
-                param->output(0).get_tensor().add_names({param_name});
-                m_inputs.emplace_back(
-                    ov::Output<const ov::Node>{param->output(0).get_node(), param->output(0).get_index()});
             }
-            for (const auto& result : model->get_results()) {
-                auto fake_param = std::make_shared<ov::op::v0::Parameter>(result->get_output_element_type(0),
-                                                                          result->get_output_partial_shape(0));
-                const std::string res_name = ov::op::util::create_ie_output_name(result->input_value(0));
+            // param->output(0).get_tensor().set_names({param_name});
```

Comment:
@riverlijunjie I comment the set_names here, otherwise the port check will fail

Reply:
If we comment the set_names, the `onnx_duplicated_output_name` can pass, but as we discussed that the graph after transformation still be different with  the master branch's result during we run below test case:
./ov_onnx_frontend_tests --gtest_filter=INTERPRETER.onnx_duplicated_output_name

1. original model:
![image](https://github.com/openvinotoolkit/openvino/assets/31196718/fcdf5c1d-1c2e-476c-bca4-2614a3329dc4)

2. master branch's transformation result:

![image](https://github.com/openvinotoolkit/openvino/assets/31196718/588e7b88-e68f-4673-aaa1-1c2afe8f540d)


3. This pr's transformation results:
![image](https://github.com/openvinotoolkit/openvino/assets/31196718/48a050db-57c3-4af4-a981-b6696341b172)